compiler/rustc_attr_parsing/src/attributes/cfg_old.rs

```diff
@@ -0,0 +1,247 @@
+use rustc_ast::{LitKind, MetaItem, MetaItemInner, MetaItemKind, MetaItemLit, NodeId};
```

Comment:
This file is temporarily still needed for the following things: `#[doc]`, `#[link]`, `cfg!()`, `#[cfg_attr]`.
In the future this file, as well as a few others, should no longer be needed, but making these attributes use the new cfg parser is sadly not easily possible without parsing the entire attributes.

As well as this file, there is also the `Cfg` type in rustdoc and all errors associated with this file. All in all this PR could have a significantly negative diff (all of this is like ~600 lines of code) if not for this